CS4850Group5A/VirtuDoc
# Email link helper

# Summary
- Added EmailDialect with link processor.
- Changed welcome email to use link helper with site-specific links.

Closes #185

# How to Test
Launch the web app and register for an account. Check SMTP4Dev and verify that the images appear correctly and that the links correctly redirect to the locally-running version of the web app. (Note that the images will not show up if you have not properly set up the self-signed SSL certificate as described here: https://github.com/CS4850Group5A/VirtuDoc/wiki/Setting-up-SSL)

# Comments
This PR creates a Thymeleaf dialect that automatically replaces links in emails with fully-qualified URLs based on what environment the web app is running in. See `src/main/resources/templates/test/linkinjection.html` and `src/main/resources/templates/test/imginjection.html` for a minimal implementation using both `<a>` and `<img>` HTML tags.

Tagging both @mo-umar and @aaraujo97 for review since this PR modifies the welcome email that they were just working on (anyone else can also review, but I'm looking for approval from at least these two before merging). Just want to make sure that they are aware of the change and how the helper library is used inside of an email template.

Feel free to leave suggestions, ideas, and criticisms in your review if you have a different idea on how to implement #185.
